src/ast/passes/codegen_llvm.cpp

```diff
@@ -986,9 +987,9 @@ void CodegenLLVM::visit(Call &call)
       buf = b_.CreateAllocaBPF(array_t, "addr4");
     }
 
-    char dst[addr_size];
+    char *dst = (char *)alloca(addr_size);
```

Comment:
```suggestion
    std::vector<uint8_t> dst(addr_size);
```

`alloca` is a very C-style way of doing things. I think the best way in C++ would be to just create a vector of the desired length.

(same comment applies to the other instances)

Reply:
I know & thought about that; eventually I ended up using alloca because I decided to stay as close to the original semantic as possible. If you want me to rewrite things with vector (which will imply a heap allocation) I'm fine with that; I don't think any of these things are on super-performance-critical paths.

Reply:
Yeah, this code is just run during compilation so isn't performance-critical (although performance is nice!). Tbh I hadn't read this code too carefully before posting my comment. It was more of an instinctive reaction to seeing manual memory allocation which I'm trying to stamp out :) 

I suppose `char[16]` would also work since we know the maximum size at compile time, although I'm not sure the performance is worth the increased risks of bugs in the future from decoupling the array length from the `addr_size` variable.

Reply:
Replaced alloca with vectors.